puppet/lib/puppet/parser/functions/query.rb

```diff
@@ -0,0 +1,45 @@
+Puppet::Parser::Functions.newfunction(:query, :type => :rvalue, :arity => -3, :doc => <<-EOT
+Query PuppetDB
+
+Usage:
+  query(["from", "resources",
+         ["and", ["=", "type", "Class"],
+          ["=", "title", "Puppet_enterprise::Profile::Puppetdb"],
+          ["not", ["=", "certname", $certname]]]])
+  [{"certname" => "foo.bar.com"
+    "environment" => "production",
+    "parameters" => {"master_certname" => "master.foo.lan"
+                     ...}} ...]
+
+  query(["from", "resources",
+         ["and", ["=", "type", "Class"],
+          ["=", "title", "Puppet_enterprise::Profile::Puppetdb"],
+          ["not", ["=", "certname", $certname]]]],
+        'certname')
+  {"foo.bar.com" => {"certname" => "foo.bar.com"
+                     "parameters" => {"master_certname" => "master.foo.lan"
+                                      ...}}}
+
+  query(["from", "resources",
```

Comment:
Same here, just feels like we should keep what the function does for now very simple, and add this later on when demand hits, or we feel its necessary. Again, this kind of thing should be a generic function also, probably in stdlib or something?